kafka-ui-react-app/src/components/Topics/Details/Messages/MessageItem.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { format } from 'date-fns';
+import JSONTree from 'react-json-tree';
+import { TopicMessage } from 'generated-sources';
+
+interface MessageItemProp {
+  partition: TopicMessage['partition'];
+  offset: TopicMessage['offset'];
+  timestamp: TopicMessage['timestamp'];
+  content: Record<string, unknown>;
+}
+
+const MessageItem: React.FC<MessageItemProp> = ({
+  partition,
+  offset,
+  timestamp,
+  content,
+}) => {
+  const getFormattedDate = (date: Date) => {
```

Comment:
I would suggest moving it out of the component. It could be a separate `util` file in case it could be used in other places in advance or this file outside of the component. After that, you could use `useMemo` to calculate it once.